### PR TITLE
23 create comment app

### PR DESCRIPTION
**Pull request by yuriiZherebukh**
## TRELLO

* [Main Trello ticket](https://trello.com/b/BJDdC443/mytrip)

## Code reviewers

- [x] @Ashiato
- [ ] @black-art
- [x] @hrytskivr
- [x] @KarnaushenkoVO
- [x] @PanLeon

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

Created relations one-to-many to Trip, Checkpoint, Photo models.
Added methods to get QuerySet from Trip, Checkpoint, Photo id's.
Updated model methods.
Added routes to Trip, Checkpoint, Photo urls.
Updated view methods.
Added docstrings

## CHECK LIST

- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/comment/views.py [hunk 1]
```diff
@@ -10,27 +10,43 @@
...
+            JsonResponse: response: <comment>
             or
-            HttpResponse:
-                {
-                    status: 404
-                }
+            HttpResponse: status: 404.
         """
+        if trip_id:
+            if checkpoint_id:
```
Comment (hrytskivr) [inline myTrip/comment/views.py]: group with 'and'

File: myTrip/comment/models.py [hunk 2]
```diff
@@ -22,56 +38,134 @@ def get_by_id(comment_id):
...
+        """
+        Get Comments with given trip id
+        Args:
+            checkpoint_id (int): user id foreign key to Trip.
+        Returns:
+            QuerySet<Comment>: QuerySet of Comments.
+        """
+        try:
+            return Comment.objects.filter(checkpoint=checkpoint_id)
+        except ObjectDoesNotExist:
```
Comment (KarnaushenkoVO) [inline myTrip/comment/models.py]: If there are no elements 'filter' will return empty QuerySet, but it never raises 'ObjectDoesNotExist'

File: myTrip/checkpoint/urls.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
-from django.conf.urls import url
+from django.conf.urls import url, include
 from .views import CheckpointView
```
Comment (hrytskivr) [inline myTrip/checkpoint/urls.py]: add a space before different import source

File: myTrip/comment/models.py [hunk 4]
```diff
@@ -1,19 +1,35 @@
 """This module contains comment model class and basic functions."""
 
+from datetime import datetime
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
```
Comment (hrytskivr) [inline myTrip/comment/models.py]: same here

File: myTrip/comment/models.py (hunk 2 above)
Comment (hrytskivr) [reply myTrip/comment/models.py]: good point!

File: myTrip/comment/views.py [hunk 5]
```diff
@@ -10,27 +10,67 @@
...
+            comments = [comment.to_dict() for comment in comments]
+            return JsonResponse(comments, status=200, safe=False)
+
+        if trip_id:
+            comments = Comment.get_by_trip_id(trip_id)
+            if not comments:
+                return HttpResponse(status=404)
+            comments = [comment.to_dict() for comment in comments]
+            return JsonResponse(comments, status=200, safe=False)
+
```
Comment (hrytskivr) [inline myTrip/comment/views.py]: looks better now

File: myTrip/tests/unittests/comment/test_model.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+            name='comment test',
+            user=CustomUser.objects.get(id=1),
+            trip=Trip.objects.get(id=10),
+            checkpoint=Checkpoint.objects.get(id=20),
+            photo=Photo.objects.get(id=30),
+            created=datetime(2017, 7, 17, 21, 00)
+        )
+
+    def test_get_by_id(self):
+        """Ensure that get by id method returns specific comment using id"""
```
Comment (hrytskivr) [inline myTrip/tests/unittests/comment/test_model.py]: dots in the end of the sentence :)

**Review by hrytskivr — COMMENTED** [review]
good overall

File: myTrip/comment/models.py [hunk 7]
```diff
@@ -1,19 +1,33 @@
 """This module contains comment model class and basic functions."""
 
+from datetime import datetime
+
+from checkpoint.models import Checkpoint
```
Comment (hrytskivr) [inline myTrip/comment/models.py]: organize imports

File: myTrip/comment/views.py [hunk 8]
```diff
@@ -5,32 +5,68 @@
...
+from checkpoint.models import Checkpoint
+from photo.models import Photo
 
 
 class CommentView(View):
     """Comments view handles GET, POST, PUT, DELETE requests."""
 
-    def get(self, request, comment_id):
+    def get(self, request, comment_id=None, trip_id=None, checkpoint_id=None, photo_id=None):
         """Handles GET request.
```
Comment (hrytskivr) [inline myTrip/comment/views.py:19]: add description of what this method does
example: shows a comment if there is one, or all comments when filtering

File: myTrip/comment/views.py [hunk 9]
```diff
@@ -5,32 +5,68 @@
 from django.views.generic.base import View
 
 from .models import Comment
+from registration.models import CustomUser
+from trip.models import Trip
+from checkpoint.models import Checkpoint
+from photo.models import Photo
```
Comment (hrytskivr) [inline myTrip/comment/views.py]: Sort imports
1) alphabetical
2) length

File: myTrip/comment/views.py [hunk 10]
```diff
@@ -5,32 +5,68 @@
...
+            or
+            HttpResponse: status: 404.
+        """
+        message = json.loads(request.body.decode('utf-8'))['message']
+        user_id = json.loads(request.body.decode('utf-8'))['user_id']
+        if not message:
+            return HttpResponse(status=400)
+        user = CustomUser.get_by_id(user_id)
+        trip = Trip.get_by_id(trip_id)
+        print(trip)
```
Comment (hrytskivr) [inline myTrip/comment/views.py]: remove

**Review by hrytskivr — COMMENTED** [review]
Fix imports and docstrings . Don't push with print statements.

**Review by hrytskivr — APPROVED** [review]
Holy moly

**Pull request merged**